### PR TITLE
update, remove junk, fix tests, use `gcc` for linking

### PR DESCRIPTION
**Pull request by MasonProtter**
I got this working again on version 1.7.1 (though it required using `gcc` to link rather than `LLVM_full_jll` :disappointed:), removed a bunch of stuff that was no longer in use, and fixed a few tests. Here's what I see now when I run the test suite:
```julia
     Testing Running tests...
Test Summary: | Pass  Total
Basics        |    6      6
Test Summary: | Pass  Total
Recursion     |    1      1
Test Summary: | Pass  Total
Loops         |    4      4
Test Summary: | Pass  Broken  Total
Arrays        |    2       1      3
Test Summary: | Pass  Total
BLAS          |    1      1
Test Summary: | Broken  Total
Hello World   |      1      1
     Testing StaticCompiler tests passed 
```

### CODE REVIEW

**Comment by tshort** [comment]
Thanks @MasonProtter! 

Tests on Windows fail. Windows doesn't like the `rm(path)` [here](https://github.com/MasonProtter/StaticCompiler.jl/blob/gpucompiler-basic/src/StaticCompiler.jl#L39). Locally, I commented that out as well as another one, and the first test block [here](https://github.com/MasonProtter/StaticCompiler.jl/blob/ffde87606a4e781779dc767985b38cd6bdf2b4ab/test/runtests.jl#L6-L20) fails 4/6 (Float32 and Float64 passed).

We could disable windows tests and add a note that it's broken on Windows.

**Comment by MasonProtter** [comment]
Yeah, Windows is a bit of a nightmare with this stuff. WE can just recommend people use the WSL

File: src/StaticCompiler.jl [hunk 1]
```diff
@@ -1,46 +1,64 @@
...
+    target = GPUCompiler.NativeCompilerTarget(always_inline=true)
+    params = TestCompilerParams()
+    GPUCompiler.CompilerJob(target, source, params), kwargs
+end
+
+function native_code_llvm(@nospecialize(func), @nospecialize(types); kwargs...)
+    job, kwargs = native_job(func, types; kwargs...)
+    GPUCompiler.code_llvm(stdout, job; kwargs...)
+end
+
```
Comment (tshort) [inline src/StaticCompiler.jl:37]: Export these?

Comment (MasonProtter) [reply src/StaticCompiler.jl:37]: sounds good

**Comment by tshort** [comment]
@MasonProtter, I'm happy to merge anytime. Let me know.

**Comment by MasonProtter** [comment]
Okay, let's just merge it now!

**Pull request merged**